restful.module

```diff
@@ -561,3 +564,18 @@ function restful_date_time_format_element_validate($element, &$form_state) {
     )));
   }
 }
+
+/**
+ * Implements hook_restful_resource_alter().
+ */
+function restful_restful_resource_alter(ResourceInterface &$resource) {
```

Comment:
> I think that maybe we should provide a hook - where those plugins return the class, and keep the alter as a way for implementing modules to remove/ swap some classes?

I'm not sure I follow. This alter hook is for both implementing modules that provide decorators and restful (as a module that provides decorators). Keep in mind that the composition order is important. `(render cache (rate limit (articles)))` is different from `(rate limit (render cache (articles)))`